src/Documenter.jl

```diff
@@ -428,8 +428,15 @@ function deploydocs(;
                         success(`git fetch upstream`) ||
                             error("could not fetch from remote.")
 
-                        success(`git checkout -b $branch upstream/$branch`) ||
-                            error("could not checkout remote branch.")
+                        branch_exists = success(`git checkout -b $branch upstream/$branch`)
+
+                        if !branch_exists
+                            Utilities.log("assuming $branch doesn't exist yet; creating a new one.")
+                            success(`git checkout --orphan $branch`) ||
+                                error("could not create new empty branch.")
+                            success(`git rm -rf .`) ||
+                                error("could not clean up new empty branch working tree.")
```

Comment:
We found that the empty commit made it harder to view and manipulate the git repo after. I don't think it hurts to add it in.

I wanted to be safe, but given the state of the surrounding code you're right that I can get rid of that `git rm`.